tools/clang/test/CodeGenDXIL/reflection/type_sys_dump_mat_as_array_of_vector.ll

```diff
@@ -0,0 +1,114 @@
+target datalayout = "e-m:e-p:32:32-i1:32-i8:32-i16:32-i32:32-i64:64-f16:32-f32:32-f64:64-n8:16:32:64"
+target triple = "dxil-ms-dx"
+
+;struct S2 {
+;    row_major int4x3 m;
+;	int1x3 m1[2];
+;	int3x1 m2;
+;};
+
+; Make sure cbuffer reflection is correct with High level struct stripped: [rows x <cols x float>] and High level struct stripped, one row: <cols x float>
+; CHECK: ID3D12ShaderReflectionConstantBuffer:
+; CHECK-NEXT:        D3D12_SHADER_BUFFER_DESC: Name: s
+; CHECK-NEXT:          Type: D3D_CT_CBUFFER
+; CHECK-NEXT:          Size: 160
+; CHECK-NEXT:          uFlags: 0
+; CHECK-NEXT:          Num Variables: 1
+; CHECK-NEXT:        {
+; CHECK-NEXT:          ID3D12ShaderReflectionVariable:
+; CHECK-NEXT:            D3D12_SHADER_VARIABLE_DESC: Name: s
+; CHECK-NEXT:              Size: 160
+; CHECK-NEXT:              StartOffset: 0
+; CHECK-NEXT:              uFlags: (D3D_SVF_USED)
+; CHECK-NEXT:              DefaultValue: <nullptr>
+; CHECK-NEXT:            ID3D12ShaderReflectionType:
+; CHECK-NEXT:              D3D12_SHADER_TYPE_DESC: Name: S2
+; CHECK-NEXT:                Class: D3D_SVC_STRUCT
+; CHECK-NEXT:                Type: D3D_SVT_VOID
+; CHECK-NEXT:                Elements: 0
+; CHECK-NEXT:                Rows: 1
+; CHECK-NEXT:                Columns: 21
+; CHECK-NEXT:                Members: 3
+; CHECK-NEXT:                Offset: 0
+; CHECK-NEXT:              {
+; CHECK-NEXT:                ID3D12ShaderReflectionType:
+; CHECK-NEXT:                  D3D12_SHADER_TYPE_DESC: Name: int4x3
+; CHECK-NEXT:                    Class: D3D_SVC_MATRIX_ROWS
+; CHECK-NEXT:                    Type: D3D_SVT_INT
+; CHECK-NEXT:                    Elements: 0
+; CHECK-NEXT:                    Rows: 4
+; CHECK-NEXT:                    Columns: 3
+; CHECK-NEXT:                    Members: 0
+; CHECK-NEXT:                    Offset: 0
+; CHECK-NEXT:                ID3D12ShaderReflectionType:
+; CHECK-NEXT:                  D3D12_SHADER_TYPE_DESC: Name: int1x3
+; CHECK-NEXT:                    Class: D3D_SVC_MATRIX_COLUMNS
+; CHECK-NEXT:                    Type: D3D_SVT_INT
+; CHECK-NEXT:                    Elements: 2
+; CHECK-NEXT:                    Rows: 1
+; CHECK-NEXT:                    Columns: 3
+; CHECK-NEXT:                    Members: 0
+; CHECK-NEXT:                    Offset: 64
+; CHECK-NEXT:                ID3D12ShaderReflectionType:
+; CHECK-NEXT:                  D3D12_SHADER_TYPE_DESC: Name: int3x1
+; CHECK-NEXT:                    Class: D3D_SVC_MATRIX_COLUMNS
+; CHECK-NEXT:                    Type: D3D_SVT_INT
+; CHECK-NEXT:                    Elements: 0
+; CHECK-NEXT:                    Rows: 3
+; CHECK-NEXT:                    Columns: 1
+; CHECK-NEXT:                    Members: 0
+; CHECK-NEXT:                    Offset: 148
+; CHECK-NEXT:              }
+; CHECK-NEXT:            CBuffer: s
+
+%hostlayout.s = type { %hostlayout.struct.S2 }
+
+%hostlayout.struct.S2 = type { [4 x <3 x i32>], [2 x [1 x <3 x i32>]], <3 x i32> }
```

Comment:
Above, the comment says "one row: <cols x float>", but the third matrix here is 3 rows, 1 column, in column major.  Shouldn't this be `[3 x <1 x i32>]`?  Most other cases test the `<3 x i32>`, but not `[3 x <1 x i32>]` or `[3 x [1 x i32]]`.